x/emissions/module/rewards/rewards_internal.go

```diff
@@ -594,6 +719,23 @@ func GetAllReputersOutput(
 		i++
 	}
 
+	// Add a final normalization step to ensure coefficients sum to 1.0
+	finalCoeffSum, err := alloraMath.SumDecSlice(coefficients)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "GetAllReputersOutput, err summing final coefficients")
+	}
+
+	if !finalCoeffSum.IsZero() && !finalCoeffSum.Equal(alloraMath.OneDec()) {
+		normalizedCoefficients := make([]alloraMath.Dec, len(coefficients))
+		for j := range coefficients {
+			normalizedCoefficients[j], err = coefficients[j].Quo(finalCoeffSum)
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "GetAllReputersOutput, error normalizing final coefficient %d", j)
+			}
+		}
+		coefficients = normalizedCoefficients
+	}
```

Comment:
```suggestion
	if !finalCoeffSum.Equal(alloraMath.OneDec()) {
		for j := range coefficients {
			coefficients[j], err = coefficients[j].Quo(finalCoeffSum)
			if err != nil {
				return nil, nil, errors.Wrapf(err, "GetAllReputersOutput, error normalizing final coefficient %d", j)
			}
		}
	}
```

We can reuse the `coefficients` array, and I think the !0 check is not needed